### PR TITLE
Updates AssemblyScript template and miscs

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: cli/assets/templates/assemblyscript/src/main.ts [hunk 1]
```diff
@@ -12,12 +12,12 @@ const smiley = memory.data<u8>([
...
 
 export function update (): void {
-    store<u16>(w4.DRAW_COLORS, 2);
+    store<u16>(w4.DRAW_COLORS, 1 * sizeof<u16>());
     w4.text("Hello from\nAssemblyScript!", 10, 10);
 
     const gamepad = load<u8>(w4.GAMEPAD1);
     if (gamepad & w4.BUTTON_1) {
-        store<u16>(w4.DRAW_COLORS, 4);
+        store<u16>(w4.DRAW_COLORS, 2 * sizeof<u16>());
```
Comment (aduros) [inline cli/assets/templates/assemblyscript/src/main.ts:20]: This shouldn't change I think. The value of "4" simply means the 4th palette color, and is independent of the size of u16.

File: cli/assets/templates/assemblyscript/asconfig.json [hunk 2]
```diff
@@ -5,15 +5,18 @@
   "options": {
     "binaryFile": "build/cart.wasm",
     "runtime": "incremental",
+    "sourceMap": true,
     "importMemory": true,
     "initialMemory": 1,
     "maximumMemory": 1,
     "noExportMemory": true,
+    "zeroFilledMemory": true,
```
Comment (aduros) [inline cli/assets/templates/assemblyscript/asconfig.json:13]: Thanks for implementing this new feature in AS! It's really great for WASM-4 and other projects that import memory :smile:

File: cli/assets/templates/assemblyscript/asconfig.json [hunk 3]
```diff
@@ -5,15 +5,18 @@
   "options": {
     "binaryFile": "build/cart.wasm",
     "runtime": "incremental",
+    "sourceMap": true,
```
Comment (aduros) [inline cli/assets/templates/assemblyscript/asconfig.json:8]: Should we put this only for debug builds?

File: cli/assets/templates/assemblyscript/src/wasm4.ts [hunk 4]
```diff
@@ -40,29 +40,34 @@ export const BUTTON_DOWN: u8 = 128;
 // └───────────────────────────────────────────────────────────────────────────┘
 
 /** Copies pixels to the framebuffer. */
+// @ts-ignore: decorator
```
Comment (aduros) [inline cli/assets/templates/assemblyscript/src/wasm4.ts:43]: Is this better than enabling experimentalDecorators in tsconfig.json?

File: cli/assets/templates/assemblyscript/src/wasm4.ts [hunk 5]
```diff
@@ -126,5 +138,7 @@ export function trace (str: string): void {
     const byteLength = load<u32>(changetype<usize>(str) - 4);
     traceUtf16(str, byteLength);
 }
+
+// @ts-ignore: decorator
 @external("env", "traceUtf16")
-declare function traceUtf16 (str: string, byteLength: i32): void;
+declare function traceUtf16 (str: string, byteLength: u32): void;
```
Comment (aduros) [inline cli/assets/templates/assemblyscript/src/wasm4.ts:144]: Nice catch :+1:

File: cli/assets/templates/assemblyscript/src/wasm4.ts (hunk 4 above)
Comment (eXponenta) [reply cli/assets/templates/assemblyscript/src/wasm4.ts:43]: decorating in TS not same as in AS, trappings a TS decorators willn't working for declarations.

Normal decorators working for real methods/classes and i not see any errors.

File: cli/assets/templates/assemblyscript/asconfig.json (hunk 3 above)
Comment (eXponenta) [reply cli/assets/templates/assemblyscript/asconfig.json:8]: It automatically dropped for releases.

Comment (MaxGraey) [reply cli/assets/templates/assemblyscript/asconfig.json:8]: source maps useful for all builds It's independent file. But if you have limit for bundle size better use it only debug builds

File: cli/assets/templates/assemblyscript/src/wasm4.ts (hunk 4 above)
Comment (MaxGraey) [reply cli/assets/templates/assemblyscript/src/wasm4.ts:43]: > Is this better than enabling experimentalDecorators in tsconfig.json?

TS doesn't support decorators for top-level functions. So this ignores required

File: cli/assets/templates/assemblyscript/src/main.ts (hunk 1 above)
Comment (MaxGraey) [reply cli/assets/templates/assemblyscript/src/main.ts:20]: Are you sure? palette have 4 colors. Index starts from zero, so 4th element should have 3th index, not 4th

**Pull request merged**